File: Payhub - SDK/ruby_SDK/PayHubSDK/lib/PayHubSDK/com/payhub/ws/vt/role_settings.rb
  class RoleSettings
    include JsonSerializer
    ATTRS=[:roleName,:firstDefaultScreen,:transactions,:reports,:help,:mobileVTAcces,:admin,:customer,:webPosAccess,:errors]
    attr_accessor *ATTRS
    USER_ROLE_LINK = "userRole/roles/"
    PATCH_USER_ROLE_LINK = "userRole/roles/update/"
    CREATE_USER_ROLE_LINK = "userRole/roles/create"
    def initialize

    end
  end

  class Transactions
    include JsonSerializer
    ATTRS=[:checked,:transactionOptions]
    attr_accessor *ATTRS

    def initialize

    end

  end

  class TransactionOptions
    include JsonSerializer
    ATTRS=[ :viewHostedShop,:viewSubmitBatch,:txtRefundupto,:single,:recurringBills]
    attr_accessor *ATTRS
    def initialize
    end

  end

  class Single
    include JsonSerializer
    ATTRS=[:checked,:singleOptions]
    attr_accessor *ATTRS
    def initialize
    end
  end

  class SingleOptions
    include JsonSerializer
    ATTRS=[:sales,:refund]
    attr_accessor *ATTRS
    def initialize
    end
  end

  class RecurringBills
    include JsonSerializer
    ATTRS=[:checked,:recurringBillOptions]
    attr_accessor *ATTRS
    def initialize
    end
  end

  class RecurringBillOptions
    include JsonSerializer
    ATTRS=[:edit,:add,:delete]
    attr_accessor *ATTRS
    def initialize
    end
  end

  class Reports
    include JsonSerializer
    ATTRS=[:checked,:reportOptions]
    attr_accessor *ATTRS
    def initialize
    end
  end

  class ReportOptions
    include JsonSerializer
    ATTRS=[:standard,:custom]
   attr_accessor *ATTRS
   def initialize
    end
  end

  class Standard
    include JsonSerializer
    ATTRS=[:checked,:standardOptions]
   attr_accessor *ATTRS
   def initialize
   end
  end

  class StandardOptions
    include JsonSerializer
    ATTRS=[:product,:recurrbill,:users,:transaction,:batch,:customer]
   attr_accessor *ATTRS
   def initialize
   end
  end

  class Help
    include JsonSerializer
    ATTRS=[:checked,:helpOptions]
   attr_accessor *ATTRS
   def initialize
   end
  end

  class HelpOptions
    include JsonSerializer
    ATTRS=[:tickets]
   attr_accessor *ATTRS
   def initialize
   end
  end

  class Tickets
    include JsonSerializer
    ATTRS=[:checked,:ticketsOptions]
   attr_accessor *ATTRS
   def initialize
   end
  end

  class TicketsOptions
    include JsonSerializer
    ATTRS=[:add,:edit]
   attr_accessor *ATTRS
   def initialize
   end
  end

  class MobileVTAcces
    include JsonSerializer
    ATTRS=[:checked]
   attr_accessor *ATTRS
   def initialize
   end
  end

  class Admin
    include JsonSerializer
    ATTRS=[:checked,:adminOptions]
   attr_accessor *ATTRS
   def initialize
   end
  end

  class AdminOptions
    include JsonSerializer
    ATTRS=[:webposSetUp,:hostedShopCart,:product,:users,:riskAndFraud,:general]
   attr_accessor *ATTRS
   def initialize
   end
  end

  class Product
    include JsonSerializer
    ATTRS=[:checked,:productOptions]
   attr_accessor *ATTRS
   def initialize
   end
  end

  class ProductOptions
    include JsonSerializer
    ATTRS=[:edit,:add,:delete,:bulkUpload]
   attr_accessor *ATTRS
   def initialize
   end
  end

  class Users
    include JsonSerializer
    ATTRS=[:checked,:usersOptions]
   attr_accessor *ATTRS
   def initialize
   end
  end

  class UsersOptions
    include JsonSerializer
    ATTRS=[:deleteRole,:manageRole,:editRole,:addRole,:edit,:add,:delete]
   attr_accessor *ATTRS
   def initialize
   end
  end

  class RiskAndFraud
    include JsonSerializer
    ATTRS=[:checked,:riskAndFraudOptions]
   attr_accessor *ATTRS
   def initialize
   end
  end

  class RiskAndFraudOptions
    include JsonSerializer
    ATTRS=[:riskfraudSetting,:riskfraudFlagTrn]
   attr_accessor *ATTRS
   def initialize
   end
  end

  class General
    include JsonSerializer
    ATTRS=[:checked,:generalOptions]
   attr_accessor *ATTRS
   def initialize
   end
  end

  class GeneralOptions
    include JsonSerializer
    ATTRS=[:validateDev,:thirdParyApi,:shippingTax,:branding,:generalSetting]
   attr_accessor *ATTRS
   def initialize
   end
  end

  class Customers
    include JsonSerializer
    ATTRS=[:checked,:customerOptions]
   attr_accessor *ATTRS
   def initialize
   end
  end

  class CustomerOptions
    include JsonSerializer
    ATTRS=[:add,:edit,:delete]
   attr_accessor *ATTRS
   def initialize
   end
  end

  class WebPosAccess
    include JsonSerializer
    ATTRS=[:checked,:webPosAccessOptions]
   attr_accessor *ATTRS
   def initialize
   end
  end

  class WebPosAccessOptions
    include JsonSerializer
    ATTRS=[:quitPos,:manualPriceChange,:admin]
   attr_accessor *ATTRS
   def initialize
   end
  end

  class AdminWebPos
    include JsonSerializer
    ATTRS=[:checked,:adminOptions]
    attr_accessor *ATTRS
    def initialize
    end
  end

  class AdminWebPosOptions
    include JsonSerializer
    ATTRS=[:hotKey,:deviceList,:generalSettings,:tills,:display,:customPrompt,:customMessage]
    attr_accessor *ATTRS
    def initialize
    end
  end

File: Payhub - SDK/c#_SDK/PayHubSDK/PayHubSDK/com/payhub/ws/vt/RoleSettings.cs
using PayHubSDK.com.payhub.ws.util;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Runtime.Serialization;
using PayHubSDK.com.payhub.ws.api;

namespace PayHubSDK.com.payhub.ws.vt
{
    [DataContract]
    public class RoleSettings
    {
	    public static string USER_ROLE_LINK = "userRole/roles/";
	    public static string PATCH_USER_ROLE_LINK = "userRole/roles/update/";
	    public static string CREATE_USER_ROLE_LINK = "userRole/roles/create";

        [DataMember]
        public string roleName;
        [DataMember]
        public int firstDefaultScreen;
        [DataMember]
        public Transactions transactions;
        [DataMember]
        public Reports reports;
        [DataMember]
        public Help help;
        [DataMember]
        public MobileVTAcces mobileVTAcces;
        [DataMember]
        public Admin admin;
        [DataMember]
        public Customer customer;
        [DataMember]
        public WebPosAccess webPosAccess;

        [DataMember]
        public List<Errors> errors;

        [DataContract]
        public class Transactions{
        [DataMember(Name = "checked")]
		public Boolean Checked;
            [DataMember]
		public TransactionOptions transactionOptions;
		
		public Transactions() {
			
		}
        [DataContract]
		public class TransactionOptions{
            [DataMember]
			public Boolean viewHostedShop;
            [DataMember]
            public Boolean viewSubmitBatch;
            [DataMember]
            public float txtRefundupto;
            [DataMember]
            public Single single;
            [DataMember]
            public RecurringBills recurringBills;
			public TransactionOptions() {
				
			}
            [DataContract]
			public class Single{
                [DataMember(Name = "checked")]
                public Boolean Checked;
                [DataMember]
                public SingleOptions singleOptions;
				
				public Single() {
					
				}
                [DataContract]
				public class SingleOptions{
                    [DataMember]
                    public Boolean sales;
                    [DataMember]
                    public Boolean refund;										
				}
			}
			
            [DataContract]
			public class RecurringBills{
                [DataMember(Name = "checked")]
                public Boolean Checked;
                [DataMember]
                public RecurringBillOptions recurringBillOptions;
				public RecurringBills() {
					
				}
                [DataContract]
				public class RecurringBillOptions{

                    [DataMember]
                    public Boolean edit;
                    [DataMember]
                    public Boolean add;
                    [DataMember]
                    public Boolean delete;
				}
			}
		}
	}
	[DataContract]
	public class Reports{
        [DataMember(Name = "checked")]
        public Boolean Checked;
        [DataMember]
        public ReportOptions reportOptions;	

	    [DataContract]
		public class ReportOptions{
            [DataMember]
            public Standard standard;
            [DataMember]
            public Boolean custom;
			
            [DataContract]
			public class Standard{
                [DataMember(Name = "checked")]
                public Boolean Checked;
                [DataMember]
                public StandardOptions standardOptions;
				
                [DataContract]
				public class StandardOptions{
                    [DataMember]
                    public Boolean product;
                    [DataMember]
                    public Boolean recurrbill;
                    [DataMember]
                    public Boolean users;
                    [DataMember]
                    public Boolean transaction;
                    [DataMember]
                    public Boolean batch;
                    [DataMember]
                    public Boolean customer;					
				}
			}
		}
	}
	[DataContract]
	public class Help{
        [DataMember(Name = "checked")]
        public Boolean Checked;
        [DataMember]
        public HelpOptions helpOptions;
		
        [DataContract]
		public class HelpOptions{
            [DataMember]
            public Tickets tickets;
			
            [DataContract]
			public class Tickets{
                [DataMember(Name = "checked")]
                public Boolean Checked;
                [DataMember]
                public TicketsOptions ticketsOptions;
				
                [DataContract]
				public class TicketsOptions{
                    [DataMember]
                    public Boolean add;
                    [DataMember]
                    public Boolean edit;					
				}
            }
		}
	}
	[DataContract]
	public class MobileVTAcces{
        [DataMember(Name = "checked")]
        public Boolean Checked;		
	}
	
    [DataContract]
	public class Admin{
        [DataMember(Name = "checked")]
        public Boolean Checked;
        [DataMember]
        public AdminOptions adminOptions;
		
        [DataContract]
		public class AdminOptions{
            [DataMember]
            public Boolean webposSetUp;
            [DataMember]
            public Boolean hostedShopCart;
            [DataMember]
            public Product product;
            [DataMember]
            public Users users;
            [DataMember]
            public RiskAndFraud riskAndFraud;
            [DataMember]
            public General general;
			
            [DataContract]
			public class Product{
                [DataMember(Name = "checked")]
                public Boolean Checked;
                [DataMember]
                public ProductOptions productOptions;
				
                [DataContract]
				public class ProductOptions{
                    [DataMember]
                    public Boolean edit;
                    [DataMember]
                    public Boolean add;
                    [DataMember]
                    public Boolean delete;
                    [DataMember]
                    public Boolean bulkUpload;					
				}
			}
            [DataContract]
			public class Users{
                [DataMember(Name = "checked")]
                public Boolean Checked;
                [DataMember]
                public UsersOptions usersOptions;
				
                [DataContract]
				public class UsersOptions{
                    [DataMember]
                    public Boolean deleteRole;
                    [DataMember]
                    public Boolean manageRole;
                    [DataMember]
                    public Boolean editRole;
                    [DataMember]
                    public Boolean addRole;
                    [DataMember]
					public Boolean edit;
                    [DataMember]
                    public Boolean add;
                    [DataMember]
                    public Boolean delete;				
				}
			}
			[DataContract]
			public class RiskAndFraud{
                [DataMember(Name = "checked")]
                public Boolean Checked;
                [DataMember]
                public RiskAndFraudOptions riskAndFraudOptions;
				public RiskAndFraud() {
					
				}
                [DataContract]
				public class RiskAndFraudOptions{
                    [DataMember]
                    public Boolean riskfraudSetting;
                    [DataMember]
                    public Boolean riskfraudFlagTrn;					
				}	
			}
            [DataContract]
			public class General{
                [DataMember(Name = "checked")]
                public Boolean Checked;
                [DataMember]
                public GeneralOptions generalOptions;
				public General() {
					
				}
                [DataContract]
				public class GeneralOptions{
                    [DataMember]
                    public Boolean validateDev;
                    [DataMember]
                    public Boolean thirdParyApi;
                    [DataMember]
                    public Boolean shippingTax;
                    [DataMember]
                    public Boolean branding;
                    [DataMember]
                    public Boolean generalSetting;
					public GeneralOptions() {						
					}	
				}
			}
		}
	}
	[DataContract]
	public class Customer{
        [DataMember(Name = "checked")]
        public Boolean Checked;
        [DataMember]
        public CustomerOptions customerOptions;
		
        [DataContract]
		public class CustomerOptions{
            [DataMember]
            public Boolean add;
            [DataMember]
            public Boolean edit;
            [DataMember]
            public Boolean delete;			
	    }
    }
        [DataContract]
	public class WebPosAccess{
            [DataMember(Name = "checked")]
            public Boolean Checked;
            [DataMember]
            public WebPosAccessOptions webPosAccessOptions;
		
            [DataContract]
		public class WebPosAccessOptions{
                [DataMember]
                public Boolean quitPos;
                [DataMember]
                public Boolean manualPriceChange;
                [DataMember]
                public AdminWebPos admin;

			[DataContract]
			public class AdminWebPos{
                [DataMember(Name = "checked")]
                public Boolean Checked;
                [DataMember]
                public AdminWebPosOptions adminOptions;
				
                [DataContract]
				public class AdminWebPosOptions{
                    [DataMember]
                    public Boolean hotKey;
                    [DataMember]
                    public Boolean deviceList;
                    [DataMember]
                    public Boolean generalSettings;
                    [DataMember]
                    public Boolean tills;
                    [DataMember]
                    public Boolean display;
                    [DataMember]
                    public Boolean customPrompt;
                    [DataMember]
                    public Boolean customMessage;				
				}
				
			}
		}
	    }
    }
    
}
